dotnet/dex-agent/GitHubPlugin.cs

```diff
@@ -0,0 +1,174 @@
+                                {
+                                    labels.Add(label.Name);
+                                }
+                            }
+                        }
+
+                        if (pr.Assignees != null)
+                        {
+                            foreach (var assignee in pr.Assignees)
+                            {
+                                if (!string.IsNullOrEmpty(assignee.Login))
+                                {
+                                    assignees.Add(assignee.Login);
+                                }
+                            }
+                        }
+                    }
+
+                    var card = GitHubCards.CreateListPRsAdaptiveCard("Pull Requests", currPullRequests, labels, assignees, authors);
+                    var attachment = new Attachment
+                    {
+                        ContentType = AdaptiveCard.ContentType,
+                        Content = card
+                    };
+                    var activity = MessageFactory.Attachment(attachment);
+                    activity.Entities.Add(new Entity
+                    {
+                        Type = "https://schema.org/Message",
+                        Properties = JObject.FromObject(new Dictionary<string, object>
+                           {
+                               { "@type", "Message" },
+                               { "@context", "https://schema.org" },
+                               { "additionalType", new List<string> { "AIGeneratedContent" } }
+                           })
+                    });
+                    await context.SendActivityAsync(activity);
+                    return JsonConvert.SerializeObject(activity);
+                }
+                else
+                {
+                    string errorBody = await response.Content.ReadAsStringAsync();
+                    Console.Error.WriteLine($"GitHub API Error Details: Status: {response.StatusCode}, " +
+                                           $"Reason: {response.ReasonPhrase}, Body: {errorBody}");
+
+                    throw new Exception($"GitHub API returned {response.StatusCode}: {errorBody}");
+                }
+            }
+            catch (Exception)
+            {
+                throw new Exception("Error accessing GitHub API");
+            }
+        }
+
+        /// <summary>
+        /// Filters the pull requests based on labels, assignees, and authors.
+        /// </summary>
+        /// <param name="labels">The labels used to filter</param>
+        /// <param name="assignees">The assignees used to filter</param>
+        /// <param name="authors">The authors used to filter</param>
+        /// <param name="context">The turn context</param>
+        /// <param name="pullRequests">The list of pull requests</param>
+        /// <returns></returns>
+        [KernelFunction, Description("Filters the pull requests")]
+        public async Task<string> FilterPRs(
```

Comment:
I imagine this being extended filter based on PR description as well. Ex. "Which PR introduced feature X".

Reply:
This one is a specific filter that shows up as toggles (can check README GIF for reference)